src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs

```diff
@@ -79,6 +82,25 @@ public static class OpenTelemetryLoggingExtensions
                     disposeProvider: false)));
 
         return builder;
+
+        // The warning here is about the fact that the OpenTelemetryLoggerOptions will be bound to configuration using ConfigurationBinder
+        // That uses reflection a lot - so if any of the properties on that class were complex types reflection would be used on them
+        // and nothing could guarantee its correctness.
+        // Since currently this class only contains primitive (boolean) properties this is OK. The top level properties are kept
+        // because the first generic argument is annotated with DAM(All) so will preserve everything on the OpenTelemetryLoggerOptions.
+        // But it would not work recursively into complex property values;
+        // This should be fully fixed with the introduction of Configuration binder source generator in .NET 8
+        // and then there should be a way to do this without any warnings.
+        // The correctness of these suppressions is verified by a test which validates that all properties of OpenTelemetryLoggerOptions
+        // are of a primitive type.
+#if NET6_0_OR_GREATER
+        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "OpenTelemetryLoggerOptions only contains boolean properties.")]
+        [UnconditionalSuppressMessage("AOT", "IL3050", Justification = "OpenTelemetryLoggerOptions only contains boolean properties.")]
+#endif
+        static void RegisterLoggerProviderOptions(ILoggingBuilder builder)
```

Comment:
This code only needs the `ServiceCollection` of the `builder`. Could you update the method signature to take in `ServiceCollection` instead?